src/Segmentation/Watersheds/SegmentWithWatershedAndDistanceMap/segmentation_result_visualization.py

```diff
@@ -0,0 +1,136 @@
+args = parser.parse_args()
+
+
+stack_image = itk.imread(args.input_image_filename)
+
+# Display the input image projections
+fig, (ax1, ax2, ax3) = plt.subplots(1, 3, figsize=(12, 4))
+for i, (cax, clabel) in enumerate(zip([ax1, ax2, ax3], ["xy", "zy", "zx"])):
+    cax.imshow(np.sum(stack_image,i).squeeze(), cmap="bone_r")
+    cax.set_title("{} Projection".format(clabel))
+    cax.set_xlabel(clabel[0])
+    cax.set_ylabel(clabel[1])
+
+plt.save(args.input_projections_output_filename)
+
+bubble_image = itk.imread(args.bubble_filename)
+
+plt.imshow(bubble_image[5], cmap="bone")
+plt.save(args.bubble_filename)
+
+# Cut-through: display the segmentation pipeline image results
+# Show the values at a slice in the middle as a way to get a feeling for what
+# the distance map and the watershed did
+
+ws_vol = itk.imread(args.watershed_filename)
+
+dmap_vol = itk.imread(args.distance_map_filename)
+
+mid_slice = ws_vol.shape[0] // 2
+fig, (ax1, ax2, ax3) = plt.subplots(1, 3, figsize=(20, 7))
+ax1.imshow(bubble_image[mid_slice], cmap="bone")
+ax1.set_title("Bubble Image")
+m_val = np.abs(dmap_vol[mid_slice]).std()
+ax2.imshow(dmap_vol[mid_slice], cmap="RdBu", vmin=-m_val, vmax=m_val)
+ax2.set_title("Distance Image\nMin:{:2.2f}, Max:{:2.2f}, Mean:{:2.2f}".format(
+    dmap_vol[mid_slice].min(), dmap_vol[mid_slice].max(), dmap_vol[mid_slice].mean()))
+ax3.imshow(ws_vol[mid_slice], cmap="nipy_spectral")
+ax3.set_title("Watershed\nLabels Found:{}".format(len(np.unique(ws_vol[ws_vol > 0]))))
+
+plt.save(args.pipeline_images_output_filename)
+
+# Show segmentation result projections
+fig, (ax1, ax2, ax3) = plt.subplots(1, 3, figsize=(12, 4))
+for i, (cax, clabel) in enumerate(zip([ax1, ax2, ax3], ["xy", "zy", "zx"])):
+    cax.imshow(np.max(ws_vol, i).squeeze(), cmap="nipy_spectral")
+    cax.set_title("{} Projection".format(clabel))
+    cax.set_xlabel(clabel[0])
+    cax.set_ylabel(clabel[1])
+
+plt.save(args.segmentation_projections_output_filename)
+
+# Clean segmentation result projections
+bubble_label_image = itk.imread(args.clean_segmentation_filename)
+
+fig, (ax1, ax2, ax3) = plt.subplots(1, 3, figsize=(12, 4))
+for i, (cax, clabel) in enumerate(zip([ax1, ax2, ax3], ["xy", "zy", "zx"])):
+    cax.imshow(np.max(bubble_label_image, i), cmap=plt.cm.jet, vmin=0, vmax=new_idx)
+    cax.set_title("{} Projection".format(clabel))
+    cax.set_xlabel(clabel[0])
+    cax.set_ylabel(clabel[1])
+
+plt.save(args.clean_segmentation_projections_output_filename)
+
+# Show 3D Rendering
+from mpl_toolkits.mplot3d.art3d import Poly3DCollection
+from skimage import measure
+from tqdm import tqdm
+def show_3d_mesh(image, thresholds):
+    p = image[::-1].swapaxes(1, 2)
+    cmap = plt.cm.get_cmap("nipy_spectral_r")
+    _fig = plt.figure(figsize=(10, 10))
+    ax = _fig.add_subplot(111, projection="3d")
+    for _i, c_threshold in tqdm(list(enumerate(thresholds))):
+        verts, faces, _, _ = measure.marching_cubes(p, level=c_threshold, 0)
```

Comment:
Having a named key for the second argument but not the subsequent argument could be a problem.

Reply:
True :+1:. Changed in 3fbe489.